pandas/core/groupby/groupby.py

```diff
@@ -3220,17 +3223,6 @@ def _nth(
                 values = np.where(nulls, NA, grouper)  # type: ignore[call-overload]
                 grouper = Index(values, dtype="Int64")
 
-        else:
-            # create a grouper with the original parameters, but on dropped
-            # object
-            grouper, _, _ = get_grouper(
-                dropped,
-                key=self.keys,
-                axis=self.axis,
-                level=self.level,
-                sort=self.sort,
-            )
```

Comment:
When keys is not a BaseGrouper, this would create a new grouper and the caches would then be recomputed meaning no time is really saved. When keys is a BaseGrouper and a null row is dropped the nth operation would raise.